### PR TITLE
test(gcp): Add new services tests to GCP

### PR DESCRIPTION
**Pull request by puchy22**
### Context

Every GCP service was not tested.

- [x] apikeys
- [x] artifacts
- [x] bigquery
- [x] cloudresourcemanager
- [x] cloudsql
- [x] cloudstorage
- [x] compute
- [x] dataproc
- [x] dns
- [x] gcr
- [x] gke
- [x] iam
- [x] kms
- [x] logging
- [x] monitoring
- [x] serviceusage

### Description

The new tests with the API calls mocked.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/gcp/services/iam/iam_service.py [hunk 1]
```diff
@@ -117,12 +117,11 @@ def __get_settings__(self):
                         name=f"projects/{project_id}/accessApprovalSettings"
                     )
                 ).execute()
-                self.settings[project_id].append(
-                    Setting(
-                        name=response["name"],
-                        project_id=project_id,
-                    )
+                self.settings[project_id] = Setting(
```
Comment (jfagoagas) [inline prowler/providers/gcp/services/iam/iam_service.py:120]: Why this change?

File: tests/providers/gcp/services/cloudresourcemanager/cloudresourcemanager_service_test.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_CloudResourceManager_Service:
+    def test__get_service__(self):
```
Comment (jfagoagas) [inline tests/providers/gcp/services/cloudresourcemanager/cloudresourcemanager_service_test.py]:
```suggestion
    def test_service(self):
```

File: tests/providers/gcp/services/cloudresourcemanager/cloudresourcemanager_service_test.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+)
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_CloudResourceManager_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/cloudresourcemanager/cloudresourcemanager_service_test.py]:
```suggestion
class TestCloudResourceManagerService:
```

File: tests/providers/gcp/services/bigquery/bigquery_service_test.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+from prowler.providers.gcp.services.bigquery.bigquery_service import BigQuery
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_BigQuery_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/bigquery/bigquery_service_test.py]:
```suggestion
class TestBigQueryService:
```

File: tests/providers/gcp/services/apikeys/apikeys_service_test.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from prowler.providers.gcp.services.apikeys.apikeys_service import APIKeys
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_APIKeys_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/apikeys/apikeys_service_test.py]:
```suggestion
class TestAPIKeysService:
```

File: tests/providers/gcp/services/bigquery/bigquery_service_test.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            assert bigquery_client.datasets[1].id.__class__.__name__ == "str"
+            assert bigquery_client.datasets[1].region == "EU"
+            assert not bigquery_client.datasets[1].cmk_encryption
+            assert not bigquery_client.datasets[1].public
+            assert bigquery_client.datasets[1].project_id == GCP_PROJECT_ID
+
+            assert len(bigquery_client.tables) == 2
+
+            assert bigquery_client.tables[0].name == "unique_table1_name"
+            assert bigquery_client.tables[0].id.__class__.__name__ == "str"
```
Comment (jfagoagas) [inline tests/providers/gcp/services/bigquery/bigquery_service_test.py:46]: Why are you doing this assert instead of the value?

File: tests/providers/gcp/services/cloudsql/cloudsql_service_test.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+from prowler.providers.gcp.services.cloudsql.cloudsql_service import CloudSQL
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_CloudSQL_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/cloudsql/cloudsql_service_test.py]:
```suggestion
class TestCloudSQLService:
```

File: tests/providers/gcp/services/cloudstorage/cloudstorage_service_test.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+)
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_CloudStorage_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_service_test.py]:
```suggestion
class TestCloudStorageService:
```

File: tests/providers/gcp/services/compute/compute_service_test.py [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+from prowler.providers.gcp.services.compute.compute_service import Compute
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_Compute_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/compute/compute_service_test.py]:
```suggestion
class TestComputeService:
```

File: tests/providers/gcp/services/dataproc/dataproc_service_test.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_Dataproc_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/dataproc/dataproc_service_test.py]:
```suggestion
class TestDataprocService:
```

File: tests/providers/gcp/services/dns/dns_service_test.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+from prowler.providers.gcp.services.dns.dns_service import DNS
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_DNS_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/dns/dns_service_test.py]:
```suggestion
class TestDNSService:
```

File: tests/providers/gcp/services/gke/gke_service_test.py [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+from prowler.providers.gcp.services.gke.gke_service import GKE
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_GKE_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/gke/gke_service_test.py]:
```suggestion
class TestGKEService:
```

File: tests/providers/gcp/services/iam/iamgcp_service_test.py [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+)
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_IAM_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/iam/iamgcp_service_test.py]:
```suggestion
class TestIAMService:
```

File: tests/providers/gcp/services/iam/iamgcp_service_test.py [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+                2021, 1, 1, 0, 0, 0
+            )
+            assert iam_client.service_accounts[1].keys[0].valid_before == datetime(
+                2022, 1, 1, 0, 0, 0
+            )
+            assert iam_client.service_accounts[1].keys[0].origin == "USER_PROVIDED"
+            assert iam_client.service_accounts[1].keys[0].type == "KEY_TYPE_UNSPECIFIED"
+
+
+class Test_AccessApproval:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/iam/iamgcp_service_test.py]:
```suggestion
class TestAccessApproval:
```

File: tests/providers/gcp/services/iam/iamgcp_service_test.py [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+                access_approval_client.settings[GCP_PROJECT_ID].name
+                == "projects/123/accessApprovalSettings"
+            )
+            assert (
+                access_approval_client.settings[GCP_PROJECT_ID].project_id
+                == GCP_PROJECT_ID
+            )
+
+
+class Test_EssentialContacts:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/iam/iamgcp_service_test.py]:
```suggestion
class TestEssentialContacts:
```

File: tests/providers/gcp/services/kms/kmsgcp_service_test.py [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+from prowler.providers.gcp.services.kms.kms_service import KMS
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_KMS_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/kms/kmsgcp_service_test.py]:
```suggestion
class TestKMSService:
```

File: tests/providers/gcp/services/logging/logging_service_test.py [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+from prowler.providers.gcp.services.logging.logging_service import Logging
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_Logging_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/logging/logging_service_test.py]:
```suggestion
class TestLoggingService:
```

File: tests/providers/gcp/services/monitoring/monitoring_service_test.py [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+from prowler.providers.gcp.services.monitoring.monitoring_service import Monitoring
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_Monitoring_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/monitoring/monitoring_service_test.py]:
```suggestion
class TestMonitoringService:
```

File: tests/providers/gcp/services/serviceusage/serviceusage_test.py [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+)
+from tests.providers.gcp.gcp_fixtures import (
+    GCP_PROJECT_ID,
+    mock_api_client,
+    mock_is_api_active,
+    set_mocked_gcp_provider,
+)
+
+
+class Test_ServiceUsage_Service:
```
Comment (jfagoagas) [inline tests/providers/gcp/services/serviceusage/serviceusage_test.py]:
```suggestion
class TestServiceUsageService:
```

File: tests/providers/gcp/services/iam/iam_organization_essential_contacts_configured/iam_organization_essential_contacts_configured_test.py
Comment (jfagoagas) [inline tests/providers/gcp/services/iam/iam_organization_essential_contacts_configured/iam_organization_essential_contacts_configured_test.py:1]: Many thanks for taking care of this test 👏

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's impressive @puchy22 👏 Many thanks for taking care of this and setting up all the calls for the GCP services.

I left some comments to adjust to Python naming conventions and a question. Do we have this documented in the Developer Guide?

**Comment by jfagoagas** [comment]
@sergargar are we sure of the backport of this, we'd need to change things related to the provider in v3.

File: prowler/providers/gcp/services/iam/iam_service.py (hunk 1 above)
Comment (puchy22) [reply prowler/providers/gcp/services/iam/iam_service.py:120]: This code was giving `Key Error` when I was testing it, and it does not make sense to do an append because the API always return one item with this call

File: tests/providers/gcp/services/bigquery/bigquery_service_test.py (hunk 6 above)
Comment (puchy22) [reply tests/providers/gcp/services/bigquery/bigquery_service_test.py:46]: Because it is generated randomly in the mock

**Comment by puchy22** [comment]
@jfagoagas This isn't in the developer guide yet, in my next PR I will do.

**Comment by MrCloudSec** [comment]
> @sergargar are we sure of the backport of this, we'd need to change things related to the provider in v3.

Yeah, we will need to do the backport with the needed changed.

**Review by jfagoagas — APPROVED** [review]
